packages/documentsearch/src/tokens.ts

```diff
@@ -263,7 +283,11 @@ export interface IBaseSearchProvider extends IDisposable {
    *
    * @returns A promise that resolves with a boolean indicating whether a replace occurred.
    */
-  replaceCurrentMatch(newText: string, loop?: boolean): Promise<boolean>;
+  replaceCurrentMatch(
+    newText: string,
+    loop?: boolean,
+    options?: IReplaceOptions
```

Comment:
Do you think `loop` should be part of `options`?

Reply:
I thought about it too. I am not sure. Pros:
- `replaceCurrentMatch` and `replaceAllMatches` would have more similar signatures
- `loop` is an option indeed

Cons:
- `loop` is not used in `replaceAllMatches`
- while `loop` is an option, it is not a user-facing option modifying the provider replace logic, but an a control variable in our internal logic

I imagine that in future `IReplaceOptions` could include things like: `replaceInComments?: bool` (which provider may support or not), `replaceUsagesInOtherFiles?: bool` which could be used by say an `LSPSearchProvider`. `loop` does not fit that well here.

Reply:
Yes good point, let's keep `loop` outside of `options`